src/mame/misc/orientalpearl.cpp

```diff
@@ -0,0 +1,152 @@
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_UNKNOWN )
+
+	PORT_START("IN1")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	
+	PORT_START("DSW0")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW0:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW0:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW0:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW0:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW0:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW0:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW0:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW0:8")
+
+	PORT_START("DSW1")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW1:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW1:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW1:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW1:8")
+INPUT_PORTS_END
+
+void orientalpearl_state::program_map(address_map &map)
+{
+	map(0x0000, 0xffff).rom();
+}
+
+void orientalpearl_state::io_map(address_map &map)
+{
+    map(0xfa00, 0xfa01).rw("kdc", FUNC(i8279_device::read), FUNC(i8279_device::write));
+    map(0xfb02, 0xfb03).w("psg", FUNC(ay8910_device::address_data_w));
+	
+}
+
+void orientalpearl_state::machine_start()
+{
+}
+
+void orientalpearl_state::orientalpearl(machine_config &config)
+{
+	/* basic machine hardware */
+	i8052_device &maincpu(I8052(config, "maincpu", XTAL(10'738'000)));
+	maincpu.set_addrmap(AS_PROGRAM, &orientalpearl_state::program_map);
+	maincpu.set_addrmap(AS_IO, &orientalpearl_state::io_map);
+	
+	/* Keyboard & display interface */
+	I8279(config, "kdc", XTAL(10'738'000) / 6); 
+	
+	/* sound hardware */
+	SPEAKER(config, "mono").front_center();
+	ay8910_device &psg(AY8910(config, "psg", XTAL(10'738'000) / 6));  // Divider not verified
+	psg.add_route(ALL_OUTPUTS, "mono", 1.0);
+	ym2413_device &opll(YM2413(config, "opll", 3.579545_MHz_XTAL));
+	opll.add_route(ALL_OUTPUTS, "mono", 1.0);
+	OKIM6295(config, "oki", XTAL(10'738'000) / 6, okim6295_device::PIN7_HIGH).add_route(ALL_OUTPUTS, "mono", 1.0);  // Clock frequency & pin 7 not verified
+}
+
+ROM_START( orientalpearl )
+	ROM_REGION( 0x10000, "maincpu", 0 )
+	ROM_LOAD( "w27c512.bin", 0x00000, 0x10000, CRC(8D3D1E91) SHA1(b80907df0878057a1ded8b56225059e06382b9d6) ) // main program
```

Comment:
Please use lowercase hex digits in the CRC.

Is the PCB location of the IC known?  If it is, you should use that as the extension rather than `bin`.